popde/adaptive_kde.py

```diff
@@ -117,6 +118,98 @@ def set_adaptive_parameter(self, new_alpha, new_global_bw):
         self.set_alpha(new_alpha)
 
 
+class KDEOptimization(AdaptiveBwKDE):
+    """
+    Given grid of bandwidth and alpha values 
+    find optimized values using cross 
+    valitation with figure of merit
+    based on loglikelihood criteria 
+    """
+    def __init__(self, data, weights, bandwidth_options, alpha_options , input_transf=None, stdize=False, rescale=None, backend='KDEpy', bandwidth=0.5, alpha=0.0, dim_names=None, do_fit=True):
+        self.alpha_options = alpha_options
+        self.bandwidth_options = bandwidth_options
+        super().__init__(data, weights, input_transf, stdize,
+                         rescale, backend, bandwidth, alpha, dim_names, do_fit)
+
+    def loo_cv_score(self, bandwidth_val, alpha_val):
+        loo = LeaveOneOut() #sklearn way
+        weights = self.weights
+        for train_index, test_index in loo.split(self.data):
+            train_data, test_data = self.data[train_index], self.data[test_index]
+           
+            #######There may need some hints how to use it better 
+            ### also if we need to add weight option here
+            local_weights = None
```

Comment:
I don't quite understand this comment .. it seems what you are doing is to optimize without using any weights at present, correct ? 
If so, make a comment that explains this and also a 'FIXME' comment to suggest that we might want to use the KDE data weights in future

Reply:
need more code to fix